src/scorepy/events.cpp

```diff
@@ -52,45 +38,119 @@ static const std::array<std::string, 2> EXIT_REGION_WHITELIST = {
 #endif
 };
 
-void region_end(const std::string& region_name)
+void region_begin(const std::string& region, const std::string& module,
+                  const std::string& file_name, const std::uint64_t line_number,
+                  const std::uintptr_t& identifier)
 {
-    const auto itRegion = regions.find(region_name);
-    if (itRegion != regions.end())
-    {
-        SCOREP_User_RegionEnd(itRegion->second.value);
-    }
-    else
+    auto& region_handle = regions[identifier];
+
+    if (region_handle == uninitialised_region_handle)
     {
-        static region_handle error_region;
-        static SCOREP_User_ParameterHandle scorep_param = SCOREP_USER_INVALID_PARAMETER;
-        static bool error_printed = false;
+        auto& region_name = make_region_name(module, region);
```

Comment:
Please add `const` as this shouldn't be changed

Reply:
bottom-up, vs top down ... But it won't hurt here. So I'll change that.